include/miral/miral/output.h

```diff
@@ -108,7 +108,7 @@ class Output
 
     /// A custom attribute map
     /// \remark Since MirAL 3.8
-    auto attributes_map() const -> std::map<std::string const, std::optional<std::string>>;
+    auto attributes_map() const -> std::map<std::string, std::optional<std::string>>;
```

Comment:
Removing the old signature is an ABI break. We should retain that.

Reply:
How do you propose that is done? If I simply add the new one right below it FTBFS with:
```
In file included from ../src/miral/active_outputs.cpp:18:
In file included from ../src/miral/display_configuration_listeners.h:20:
../include/miral/miral/output.h:112:10: error: functions that differ only in their return type cannot be overloaded
  112 |     auto attributes_map() const -> std::map<std::string, std::optional<std::string>>;
      |          ^
../include/miral/miral/output.h:111:10: note: previous declaration is here
  111 |     auto attributes_map() const -> std::map<std::string const, std::optional<std::string>>;
      |          ^
```

Reply:
What problem did you see that motivated you to change the return type?

Reply:
#3383 basically, if I don't do each hunk of the changes in 517b4b13a1bc033fd4fd03f453157d8f88ef7389 I see a similar error as I originally reported there

Reply:
To be specific, without the `attributes_map` (`include/miral/miral/output.h` and `src/miral/output.cpp`) changes I get:
```
../src/miral/output.cpp:127:12: error: no viable conversion from returned value of type 'map<basic_string<...>, [...]>' to function return type 'map<const basic_string<...>, [...]>'
  127 |     return self->custom_attribute;
      |            ^~~~~~~~~~~~~~~~~~~~~~
/usr/bin/../include/c++/v1/map:1078:25: note: candidate constructor not viable: no known conversion from 'std::map<std::string, std::optional<std::string>>' (aka 'map<basic_string<char>, optional<basic_string<char>>>') to 'const map<const string, optional<string>> &' for 1st argument
 1078 |   _LIBCPP_HIDE_FROM_ABI map(const map& __m) : __tree_(__m.__tree_) { insert(__m.begin(), __m.end()); }
      |                         ^   ~~~~~~~~~~~~~~
/usr/bin/../include/c++/v1/map:1096:25: note: candidate constructor not viable: no known conversion from 'std::map<std::string, std::optional<std::string>>' (aka 'map<basic_string<char>, optional<basic_string<char>>>') to 'map<const string, optional<string>> &&' for 1st argument
 1096 |   _LIBCPP_HIDE_FROM_ABI map(map&& __m) _NOEXCEPT_(is_nothrow_move_constructible<__base>::value)
      |                         ^   ~~~~~~~~~
/usr/bin/../include/c++/v1/map:1106:25: note: candidate constructor not viable: no known conversion from 'std::map<std::string, std::optional<std::string>>' (aka 'map<basic_string<char>, optional<basic_string<char>>>') to 'initializer_list<value_type>' (aka 'initializer_list<pair<const std::string, std::optional<std::string>>>') for 1st argument
 1106 |   _LIBCPP_HIDE_FROM_ABI map(initializer_list<value_type> __il, const key_compare& __comp = key_compare())
      |                         ^   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/bin/../include/c++/v1/map:1034:34: note: explicit constructor is not a candidate
 1034 |   _LIBCPP_HIDE_FROM_ABI explicit map(const key_compare& __comp) _NOEXCEPT_(
      |                                  ^
/usr/bin/../include/c++/v1/map:1128:34: note: explicit constructor is not a candidate
 1128 |   _LIBCPP_HIDE_FROM_ABI explicit map(const allocator_type& __a) : __tree_(typename __base::allocator_type(__a)) {}
      |                                  ^

```

Reply:
It should be enough to change:
```c++
  return self->custom_attribute;
```
to 
```c++
   return {self->custom_attribute.begin(), self->custom_attribute.end()};
```

Reply:
Indeed that appears to do the trick